controllers/extension-certificate-service/pkg/apis/config/v1alpha1/validation.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
By convention, validation is a separate package. Move it to `pkg/config/validation/validation.go` and only act on the internal version (don't validate on a versioned API).

Reply:
Okay, I haven't added an internal version yet because I thought it'd be too much overhead for an internal configuration resource.

Anyway, I can add an internal version, validate on this but would also suggest to let the reconcilers/actuators rather work with the internal version (defaulted, validated) than the external `v1alpha1`. WDYT?

Reply:
Usually, you also have internal versions for your configuration, e.g.
- https://github.com/gardener/gardener/blob/master/pkg/controllermanager/apis/config/types.go
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/types.go
- https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/apis/config/types.go
- ...

This yields the benefit that the whole API machinery can be used with this resource as well. 

> internal configuration resource.
End-users rely and configure it, so it's also external (that's why you have the `v1alpha1` version in the first place).

> but would also suggest to let the reconcilers/actuators rather work with the internal version (defaulted, validated)

Yes, definitely, this makes sense. 👍